plugins/module_utils/redfish_utils.py

```diff
@@ -3152,3 +3152,53 @@ def set_session_service(self, sessions_config):
         if resp['ret'] and resp['changed']:
             resp['msg'] = 'Modified session service'
         return resp
+
+    def enable_secure_boot(self):
+        # This function enable Secure Boot on an OOB controller
+
+        # Get server details
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if not response["ret"]:
+            return response
+
+        # Check secure boot state
+        server_details = response["data"]
+        secure_boot_url = server_details["SecureBoot"]["@odata.id"]
+
+        response = self.get_request(self.root_uri + secure_boot_url)
+        if not response["ret"]:
+            return response
+
+        secure_boot_details = response["data"]
+
+        if secure_boot_details["SecureBootEnable"]:
+            return {
+                "ret": True,
+                "changed": False,
+                "msg": "Secure boot is already enabled"
+            }
+        else:
+            # Check server power state
+            state = server_details["PowerState"]
+
+            # If server is not powered OFF, then failing with message
+            if state != "Off":
```

Comment:
This isn't necessarily something required by everyone. Services can support modifying this property while the system is on.

Reply:
Changed as suggested. Thanks!

Reply:
Resolving this conversation. Thanks!